pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -236,22 +230,22 @@ def __run_eval_epoch_end(self, num_dataloaders, using_eval_result):
 
         if self.testing:
             if is_overridden('test_epoch_end', model=model):
-                model._current_fx_name = 'test_epoch_end'
                 if using_eval_result:
                     eval_results = self.__gather_epoch_end_eval_results(outputs)
-
+                model._current_fx_name = 'test_epoch_end'
```

Comment:
again protected and bellow one more...

Reply:
Yes, this variable is protected as no use for the user, but is was meant to be use this way.

Reply:
I thought about how maybe a function setter is better but came to the same conclusion as @tchaton this is the best approach imo